meson.build

```diff
@@ -28,57 +27,84 @@ link_args = [
   '-Wl,--file-alignment=4096',
 ]
 
-# Wine's built-in back traces only work with dwarf2 symbols
-if get_option('debug') and target_machine.system() == 'windows'
-  compiler_args += [
-    '-gstrict-dwarf',
-    '-gdwarf-2',
-  ]
-endif
-
 if get_option('build_id')
   link_args += [
     '-Wl,--build-id',
   ]
 endif
 
-if cpu_family == 'x86'
-  link_args += [
-    '-Wl,--enable-stdcall-fixup',
-    '-Wl,--add-stdcall-alias',
+if platform == 'windows'
+  compiler_args += [
+    '-DNOMINMAX',
+    '-D_WIN32_WINNT=0xa00',
   ]
+
+  # Wine's built-in back traces only work with dwarf2 symbols
+  if get_option('debug')
+    compiler_args += [
+      '-gstrict-dwarf',
+      '-gdwarf-2',
+    ]
+  endif
+
+  # Enable stdcall fixup on 32-bit
+  if cpu_family == 'x86'
+    link_args += [
+      '-Wl,--enable-stdcall-fixup',
+      '-Wl,--add-stdcall-alias',
+  ]
+  endif
+
+  if (cpu_family == 'x86_64')
+    dxvk_library_path = meson.source_root() + '/lib'
+  else
+    dxvk_library_path = meson.source_root() + '/lib32'
+  endif
+
+  lib_vulkan  = cpp.find_library('vulkan-1', dirs : dxvk_library_path)
+  lib_d3d9    = cpp.find_library('d3d9')
+  lib_d3d11   = cpp.find_library('d3d11')
+  lib_dxgi    = cpp.find_library('dxgi')
+  lib_d3dcompiler_43 = cpp.find_library('d3dcompiler_43', dirs : dxvk_library_path)
+
+  if dxvk_is_msvc
+    lib_d3dcompiler_47 = cpp.find_library('d3dcompiler')
+  else
+    lib_d3dcompiler_47 = cpp.find_library('d3dcompiler_47')
+  endif
+
+  if dxvk_is_msvc
+    res_ext = '.res'
+    wrc = find_program('rc')
+    wrc_generator = generator(wrc,
+      output    : [ '@BASENAME@' + res_ext ],
+      arguments : [ '/fo', '@OUTPUT@', '@INPUT@' ],
+    )
+  else
+    res_ext = '.o'
+    wrc = find_program('windres')
+    wrc_generator = generator(wrc,
+      output    : [ '@BASENAME@' + res_ext ],
+      arguments : [ '-i', '@INPUT@', '-o', '@OUTPUT@' ],
+    )
+  endif
+
+  dxvk_include_path = include_directories('./include')
+else
+  lib_vulkan = cpp.find_library('vulkan')
+  lib_sdl2   = cpp.find_library('SDL2')
+  wrc        = find_program('true')
```

Comment:
what's this for, wouldn't you also need `wrc_generator`?

Reply:
Fixed